lib/funky/connections/base.rb

```diff
@@ -5,14 +5,30 @@ class Base
 
     private
 
+      def self.instrument(uri, request, &block)
+        data = {
+          request: request,
+          method: request.method,
+          request_uri: uri
+        }
+        if defined?(ActiveSupport::Notifications)
+          ActiveSupport::Notifications.instrument 'request.funky', data, &block
+        else
+          yield data
```

Comment:
🤔 why yield data here? it's going to blow up when you get down to `"Error #{response.code}: #{response.body}"`.

Reply:
I could also just change it to make it more explicit. e.g. `block.call(data)`

Reply:
you wanna know what? i read `yield data` as `return data` rather than `block.call(data)`

:|